BonusCalcListener/Gateway/Interfaces/IPayElementGateway.cs

```diff
@@ -8,6 +8,6 @@ namespace BonusCalcListener.Gateway
 {
     public interface IPayElementGateway
     {
-        Task<IEnumerable<PayElement>> GetPayElementsByWorkOrderId(string workOrderId, int payElementType);
+        Task<List<PayElement>> GetPayElementsByWorkOrderId(string workOrderId, int payElementType);
```

Comment:
Could you tell me why you're switching the IEnumerable with a List? Do we require correct ordering/positioning of elements?

Reply:
Because internally in `ProcessMessageAsync` it was converting it to `List` which duplicated the object and meant that there was no way to inspect what was done with the mock result from `GetPayElementsByWorkOrderId` (e.g. ensure that 'Cancelled' work orders are skipped).

Reply:
I see, thanks for clarifying.